File: introductionToAlgorithmCLRS/chp2/Ex_2.1/q4n_bitBinarySum.c
#include <stdio.h>
#include <math.h>

long long convert(int num) {
    long long bin = 0;
    int rem, i;
    i = 1;
    while (num != 0 ) {
        rem = num % 2;
        num /= 2;
        bin += rem * i;
        i *= 10;
    }
    return bin;
}

long long sumBin(long long bin1, long long bin2, long long sumBinary) {
    int init = 1;
    sumBinary = 0;
    int res = 0;
    while(bin1 != 0 && bin2 != 0) {
        sumBinary += ((bin1 %10 + bin2%10 + res) %2)*init;
        res = (bin1 %10 + bin2%10 + res)/2;
        bin1 /= 10;
        bin2 /= 10;
        init *= 10;
    }
    if (res != 0) {
        sumBinary += res*init;
    }

    return sumBinary;
}
int main() {
    int num1, num2;
    printf("Enter the number to be added : ");
    scanf("%d %d", &num1, &num2);

    long long num1Binary, num2Binary;

    num1Binary = convert(num1);
    num2Binary = convert(num2);

    printf("Binary of the %d is %lld.\n", num1, num1Binary);
    printf("Binary of the %d is %lld.\n", num2, num2Binary);

    long long sumBinary;
    sumBinary = sumBin(num1Binary, num2Binary, sumBinary);

    printf("sum in Binary is : %lld ", sumBinary);

}

File: introductionToAlgorithmCLRS/chp2/Ex_2.1/q4n_bitBinarySum.py
def sumBin(bin1, bin2):
    init = 1
    sumBinary = 0
    res = 0
    while (bin1 != 0 or bin2 != 0):
        sumBinary += ((bin1%10 + bin2%10 + res)%2)*init
        res = (bin1%10 + bin2%10 + res)//2
        bin1 = bin1 // 10
        bin2 = bin2 // 10
        init = init * 10
    if (res != 0):
        sumBinary += (res*init)
    return sumBinary

num1 = int(input("Enter first Element : "))
num2 = int(input("Enter second Element : "))
num1Binary = bin(num1).replace("0b", "")
num2Binary = bin(num2).replace("0b", "")
num1Binary = int(num1Binary)
num2Binary = int(num2Binary)

print("Binary of the " , num1 , "is", num1Binary, ".")
print("Binary of the " , num2 , "is", num2Binary, ".")

sumBinary = sumBin(num1Binary, num2Binary)

print("Sum in binary is ", sumBinary)
